hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -169,8 +169,10 @@ private AuthenticationStatus verifyEmptyCredentialsAndClusterName(PasswordCreden
     private ClientMessage prepareUnauthenticatedClientMessage() {
         boolean clientFailoverSupported = nodeEngine.getNode().getNodeExtension().isClientFailoverSupported();
         Connection connection = endpoint.getConnection();
-        logger.warning("Received auth from " + connection + " with clientUuid " + clientUuid
-                + " and clientName " + clientName + ", authentication failed");
+        if (logger.isFineEnabled()) {
```

Comment:
A general question really: why do we have these if checks in our code? If a logger does not have a log level enabled, it won't log it anyway.

Reply:
It's just to avoid unnecessary evaluations. (String concatenation only in this case)